File: backend/master/internal/model/instance.go
package model

import "time"

// SnellInstance 表示运行在节点上的 Snell 服务实例。
type SnellInstance struct {
	ID          uint      `gorm:"primaryKey" json:"id"`
	UserID      uint      `gorm:"index;not null" json:"user_id"`
	NodeID      uint      `gorm:"index;not null" json:"node_id"`
	Port        int       `gorm:"not null" json:"port"`
	PSK         string    `gorm:"size:255;not null" json:"psk"`
	Version     int       `gorm:"default:4" json:"version"`
	Obfs        string    `gorm:"size:64" json:"obfs"`
	ConfigPath  string    `gorm:"size:255" json:"config_path"`
	ServiceName string    `gorm:"size:128" json:"service_name"`
	Status      string    `gorm:"size:32;default:'stopped'" json:"status"`
	CreatedAt   time.Time `json:"created_at"`
	UpdatedAt   time.Time `json:"updated_at"`

	User User `json:"user,omitempty"`
	Node Node `json:"node,omitempty"`
}

File: frontend/admin/src/api/instance.ts
import request from '@/utils/request'

// Instance Interface
export interface Instance {
    id: number
    user_id: number
    node_id: number
    port: number
    psk: string
    version: number
    obfs: string
    config_path: string
    service_name: string
    status: string  // 'running' | 'stopped'
    created_at: string
    updated_at: string
    user?: {
        id: number
        username: string
    }
    node?: {
        id: number
        name: string
        endpoint: string
    }
}

// Instance Filter
export interface InstanceFilter {
    user_id?: number
    node_id?: number
    status?: number
    page?: number
    page_size?: number
}

// Create Instance Request
export interface CreateInstanceRequest {
    user_id: number
    node_id: number
    port?: number
    psk?: string
}

// Get Instance List
export function getInstanceList(params: InstanceFilter) {
    return request<{
        items: Instance[]
        total: number
        page: number
        page_size: number
    }>({
        url: '/admin/instances',
        method: 'get',
        params
    })
}

// Create Instance
export function createInstance(data: CreateInstanceRequest) {
    return request<Instance>({
        url: '/admin/instances',
        method: 'post',
        data
    })
}

// Get Instance Detail
export function getInstance(id: number) {
    return request<Instance>({
        url: `/admin/instances/${id}`,
        method: 'get'
    })
}

// Delete Instance
export function deleteInstance(id: number) {
    return request({
        url: `/admin/instances/${id}`,
        method: 'delete'
    })
}

// Restart Instance
export function restartInstance(id: number) {
    return request({
        url: `/admin/instances/${id}/restart`,
        method: 'post'
    })
}

// Update Instance Status
export function updateInstanceStatus(id: number, status: string) {
    return request({
        url: `/admin/instances/${id}/status`,
        method: 'put',
        data: { status }
    })
}

// Get Instance Config
export function getInstanceConfig(id: number) {
    return request<{ config: string }>({
        url: `/admin/instances/${id}/config`,
        method: 'get'
    })
}
